### ISSUE TITLE
Use async/await instead of generators as per Koa v2

### ISSUE THREAD

**Issue by pritchardjonathan** [tjmehta/coworkers#35]

**Comment by tjmehta** [comment]
They have only pulled that functionality into their "next" branch/version. We could maintain the same w/ coworkers until async/await is node native

**Comment by tjmehta** [comment]
<Please vote here if you want this functionality>

**Comment by DaAwesomeP** [comment]
+1

**Comment by DaAwesomeP** [comment]
Compatibility with something like [asyncawait](https://github.com/yortus/asyncawait) would be nice as well.

**Comment by DaAwesomeP** [comment]
@tjmehta you probably won't get very many votes here because the ratios of stargazers to watchers in this repo is very high.

**Comment by tjmehta** [comment]
That's true. If we can support both without having to maintain a "next" version like koa, I would be happy to take a PR. If we cannot support both I will probably still take a PR and maintain both versions. I personally don't use Babel on the server side, so I personally wouldn't implement it in until node supports async/await.

**Comment by mkozjak** [comment]
+1

**Comment by amcdnl** [comment]
@tjmehta Node supports async/await now ;P

**Comment by tjmehta** [comment]
Unstable only though right? I am willing to get a branch ready with support
if someone wants to take initiative. I am thinking I want to remove
clustering too if we make it a breaking change

**Comment by amcdnl** [comment]
It was introduced at Node 7, node is at 7.5 now so I'd say it is ;P

**Comment by tjmehta** [comment]
Odd majors are considered unstable in node, but I am cool with preparing
for 8 at this point

**Comment by amcdnl** [comment]
+1 

You mentioned removing clustering? Whats the motivation behind that?

**Comment by john24318** [comment]
+1

**Comment by amcdnl** [comment]
@tjmehta - any updates on moving this ahead now that async/await is in node 8.0?

**Comment by tjmehta** [comment]
I was thinking about this the other day. I am happy to take PRs and will
eventually get to it, however syntactic sugar leaves this low on my
development priority list. I will continue to maintain any issues people
have with coworkers, but I am just not sure I have bandwidth to take on
this task soon.
